client/gui/InterfaceBuilder.h

```diff
@@ -0,0 +1,39 @@
+/*
+* InterfaceBuilder.h, part of VCMI engine
+*
+* Authors: listed in file AUTHORS in main folder
+*
+* License: GNU General Public License v2.0 or later
+* Full text of license available in license.txt file, in main folder
+*
+*/
+
+#pragma once
+
+#include "CIntObject.h"
+
+#include "../../lib/JsonNode.h"
+
+class InterfaceBuilder: public CIntObject
+{
+public:
+	InterfaceBuilder();
+	InterfaceBuilder(const JsonNode & config);
+
+protected:
+	//must be called after adding callbacks
+	void init(const JsonNode & config);
+	
+	void addCallback(const std::string & callbackName, std::function<void(int)> callback);
+	
+	const std::shared_ptr<CIntObject> widget(const std::string &) const;
```

Comment:
In almost all usages of this method you use dynamic_pointer_cast.
Perhaps it makes sense to add another method, something like:
```
template<typename T>
const std::shared_ptr<T> widgetAs(const std::string & name) const
{
   return dynamic_pointer_cast<T>(widget(name));
}
```
And use it where applicable?

Reply:
quite good point, actually, only that function will be needed